File: content/learn/arrays-easy/02-second-largest/solution.py
def second_largest(a):
    largest = float("-inf")
    second = float("-inf")
    for x in a:
        if x > largest:
            second = largest
            largest = x
        elif x < largest and x > second:
            second = x
    return -1 if second == float("-inf") else second


n = int(input())
a = list(map(int, input().split()))
print(second_largest(a))

File: content/learn/arrays-easy/02-second-largest/solution.js
function secondLargest(a) {
  let largest = -Infinity;
  let second = -Infinity;
  for (const x of a) {
    if (x > largest) {
      second = largest;
      largest = x;
    } else if (x < largest && x > second) {
      second = x;
    }
  }
  return Number.isFinite(second) ? second : -1;
}

const n = parseInt(readline(), 10);
const a = readline().split(" ").map(Number);
print(secondLargest(a));
